components/calendar/src/astronomy.rs

```diff
@@ -0,0 +1,657 @@
+            470160, 473837, 507850, 524156, 544676, 567118, 569477, 601716, 613424, 626596, 645554,
+            664224, 671401, 694799, 704424, 708842, 709409, 709580, 727274, 728714, 744313, 764652,
+        ];
+        let expected_ephemeris = [
+            0.214169, 0.143632, 0.114444, 0.107183, 0.069498, 0.057506, 0.044758, 0.017397,
+            0.012796, 0.008869, 0.007262, 0.005979, 0.005740, 0.003875, 0.003157, 0.002393,
+            0.001731, 0.001669, 0.000615, 0.000177, 0.000101, 0.000171, 0.000136, 0.000061,
+            0.000014, 0.000276, 0.000296, 0.000302, 0.000302, 0.000675, 0.000712, 0.000963,
+            0.002913,
+        ];
+        let expected_solar_long = [
+            119.473431, 254.248961, 181.435996, 188.663922, 289.091566, 59.119741, 228.314554,
+            34.460769, 63.187995, 2.457591, 350.475934, 13.498220, 37.403920, 81.028130,
+            313.860493, 19.954430, 176.059431, 344.922951, 79.964921, 99.302317, 121.535304,
+            88.567428, 129.289884, 6.146910, 28.251993, 151.780699, 185.945867, 28.555607,
+            193.347892, 357.151254, 336.170692, 228.184879, 116.439352,
+        ];
+        let expected_lunar_long = [
+            244.853905, 208.856738, 213.746842, 292.046243, 156.819014, 108.055632, 39.356097,
+            98.565851, 332.958296, 92.259651, 78.132029, 274.946995, 128.362844, 89.518450,
+            24.607322, 53.485956, 187.898520, 320.172362, 314.042566, 145.474063, 185.030507,
+            142.189132, 253.743375, 151.648685, 287.987743, 25.626707, 290.288300, 189.913142,
+            284.931730, 152.339044, 51.662265, 26.6820606, 175.500822,
+        ];
+        let expected_next_new_moon = [
+            -214174.605828,
+            -61382.995328,
+            25495.809776,
+            49238.502448,
+            171318.435313,
+            210180.691849,
+            253442.859367,
+            369763.746413,
+            400091.578343,
+            434376.578106,
+            452627.191972,
+            470167.578360,
+            473858.853276,
+            507878.666842,
+            524179.247062,
+            544702.753873,
+            567146.513181,
+            569479.203258,
+            601727.033557,
+            613449.762129,
+            626620.369801,
+            645579.076748,
+            664242.886718,
+            671418.970538,
+            694807.563371,
+            704433.491182,
+            708863.597000,
+            709424.404929,
+            709602.082686,
+            727291.209400,
+            728737.447691,
+            744329.573999,
+            764676.191273,
+        ];
+        for i in 0..rd_vals.len() {
+            let moment = Moment::new(rd_vals[i] as f64);
+
+            // Checking ephemeris correction
+            let ephemeris = Astronomical::ephemeris_correction(moment);
+            let expected_ephemeris_value = expected_ephemeris[i];
```

Comment:
Issue, here and elsewhere: the ephemeris value can be very small; one of your expectations is `0.000061`, which is smaller than your range of 0.0001. You should either explicitly list a range of values up above in the testdata or assert that the value is within a certain percentile range, but don't use a fixed size window when the values vary so greatly in magnitude.

Reply:
Done